contribute-to-tidb/review-a-pr-stage.md

```diff
@@ -0,0 +1,233 @@
+
+### Tests
+
+A PR should be test covered, whether the tests are unit tests, integration tests, or end-to-end tests.
+
+Ensure the tests in the PR are correct, sensible, and useful.
+
+* Are they covering interesting cases?
+* Are they readable?
+* Does the PR lower overall test coverage? 
+
+Simulate cases where the code may break.
+
+* Will the tests actually fail when the code is broken?
+* If the code changes beneath them, will they start producing false positives?
+* Does each test makes simple and useful assertions?
+* Are the tests separated appropriately between different test methods?
+
+Remember that tests are also code that have to be maintained, and thus do not accept unnecessary complexity in tests.
+
+Last but not least, keep an eye on how long a single unit takes to run. That is, whether or not the CI is slowed by the tests introduced.
+
+### Naming
+
+Does the developer pick good names for everything? A good name is long enough to fully communicate what the item is or does, without being so long that it becomes hard to read.
+
+### Comments
+
+* Does the developer write clear comments in understandable English?
+* Are all of the comments actually necessary?
+
+Usually comments are useful when they explain why the code exists, while not explain what the code is doing. If the code isn't clear enough to explain itself, then the code should be made simpler.
+
+There are some exceptions; for example, regular expressions and complex algorithms often benefit greatly from comments that explain what they’re doing. But mostly comments are for information that the code itself can hardly contain, like the reason behind a decision.
+
+It can also be helpful to look at comments existing before this PR. Maybe there is a `TODO` can be removed now, a comment advising against this change being made, etc.
+
+Note that comments are different from documentation of classes, modules, or functions, which should instead express the purpose of a piece of code, how it should be used, and how it behaves when used.
+
+### Style
+
+Make sure the PR follows our [code style and quality guide](code-style-and-quality-guide.md). For Go and Rust, there are built in tools with the compiler toolchain.
+
+If you want to improve style points outside the style guide, you can comment that you think it would improve the code readability but isn't mandatory. Don't block PRs from being submitted based only on personal style preferences.
+
+The PR author should not include major style changes combined with other changes. For example, if a PR wants to reformat the whole file should not make functional changes. This has been explained as "One PR just does one thing" in the previous section.
```

Comment:
The PR author should combine major style changes -- such as reformatting the entire file -- with functional changes. This would violate the guideline covered earlier: one PR does one thing.

Reply:
```suggestion
The PR author should not combine nontrivial style changes with functional changes. This would violate the guideline covered earlier: one PR does one thing.
```